cclient/request/attach_to_tangle.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "attach_to_tangle.h"
+
+attach_to_tangle_req_t* attach_to_tangle_req_new() {
+  attach_to_tangle_req_t* req =
+      (attach_to_tangle_req_t*)malloc(sizeof(attach_to_tangle_req_t));
+  req->trunk = char_buffer_new();
+  req->branch = char_buffer_new();
+  req->mwm = ATTACH_TO_TANGLE_MIN_MWM;
+  utarray_new(req->trytes, &ut_str_icd);
+  return req;
+}
+
+void attach_to_tangle_req_free(attach_to_tangle_req_t** req) {
+  if (*req) {
+    char_buffer_free((*req)->trunk);
+    char_buffer_free((*req)->branch);
+    utarray_free((*req)->trytes);
+    free(*req);
+    *req = NULL;
+  }
+}
+
+retcode_t attach_to_tangle_req_set_trunk(attach_to_tangle_req_t* req,
```

Comment:
could you use address_t and memcpy?

Reply:
try to keep plain text in the JSON serialization & test cases, so we can do the flex trits conversion as needed. I'm not sure about keeping the flex_tirits in a structure is better, but I will figure it out later.

Reply:
can you do that and using `cJSON_CreateArray` instead of `cJSON_CreateString` ?